src/os/bluestore/BlueFS.cc

```diff
@@ -1773,8 +1773,12 @@ int64_t BlueFS::_read_random(
 	      << dendl;
 
       // NOTE: h->bl is normally a contiguous buffer so c_str() is free.
+      s_lock.unlock();
+      std::unique_lock u_lock(h->lock);
```

Comment:
It looks like you might want to use `iterator_impl::copy(unsigned len, char *dest);` instead.
This wouldn't rebuild bufferlist and hence one is able to proceed with already acquired shared_lock..

Reply:
yep, correct. I will verify it in my environment first. Thanks!

Reply:
> It looks like you might want to use `iterator_impl::copy(unsigned len, char *dest);` instead.
> This wouldn't rebuild bufferlist and hence one is able to proceed with already acquired shared_lock..

modified by using `iterator_impl::copy(unsigned len, char *dest);`